Improve layout of the docs

agroutsmith opened the pull request:
closes #181 
- Changing the contents page to include Interrupted Time Series (examples.rst) and moving notebooks its_covid, sc2_pymc and sc2_skl into this section.
-  Distinguishing interrupted time series from synthetic controls in notebook description of sc2_pymc and sc2_skl.
-  Explicitly addressing DiD 2x2 vs multiple observations DiD in did_pymc_banks notebook.

drbenvincent reviewed (changes requested):
Great stuff! Just a few small requests:

Now we have a dedicated `InterruptedTimeSeries` class. So we could use that class, instead of in `sc2_pymc.ipynb` and `sc2_skl.ipynb`. (It already seems to be done in `its_covid.ipynb` I think because `InterruptedTimeSeries` existed in the past but got removed.)

I'd also then re-run those notebooks to check _and_ because there should be subtle changes to the plot legend and the text summary output.

For the main notebook titles can we change "Example 2..." to "Example...". This is for `sc2_pymc.ipynb` and `sc2_skl.ipynb`

Can we also change the filenames `sc2_pymc.ipynb` and `sc2_skl.ipynb` to `its_pymc.ipynb` and `its_skl.ipynb`.

agroutsmith commented:
- Changed to use ITS class
- Re-ran notebooks with ITS class - the only change I could see was on the results summary
- Renamed sc2 notebooks
- Renamed titles of sc2 notebooks

drbenvincent commented:
Hi @agroutsmith. I re-ran the notebooks. I also realised that we needed to add info about interrupted time series to the `README.md` (for GitHub) and `index.rst` (readthedocs homepage).

We have a notebook called `generate_plots.ipynb` which I used to add the (pymc) interrupted time series figure. But I realised that this notebook doesn't generate any of the scikit-learn images. I think I'll set up a new issue for that.

So I think we might be done on this PR :) Let me know if you think there's anything else.
